doc/user_guide/edml_user_guide.md

```diff
@@ -27,9 +27,108 @@ This guide explains how to define EDML mappings in general. For data source spec
+* [Google Cloud Storage](https://github.com/exasol/google-cloud-storage-document-files-virtual-schema/blob/main/doc/user_guide/user_guide.md#mapping-multiple-files)
+
+### Source Reference Column
+
+Some dialects support reading one table from multiple sources. For example the [files-virtual-schemas](https://github.com/exasol/virtual-schema-common-document-files) allow you to load each row from a different file. In that case you may want to add the filename as a column to the Exasol table. That allows you to query on it and by that only read the required files.
+
+To do so, set `"addSourceReferenceColumn": true` in the root object of your EDML definition. The adapter will then automatically add a column named `SOURCE_REFERENCE` to the end of the table:
+
+```json
+{
+  "$schema": "https://schemas.exasol.com/edml-2.1.0.json",
+  "source": "data/*.csv",
+  "destinationTable": "BOOKS",
+  "addSourceReferenceColumn": true
+}
+```
+
+You can use this property for all dialects. Typically, it will, however, only give you additional information, if you load data from multiple sources.
+
+The `SOURCE_REFERENCE` column has a maximum size of 2000 characters. The adapter will throw an exception when a source reference exceeds this.
+
+## Define Mapping of Fields to Table Columns
+
+EDML allows you to specify custom mapping from fields in source files to Exasol table columns. Virtual schema support two options for the mapping:
+* [Automatic Mapping Inference](#automatic-mapping-inference)
+* [Explicit Mapping Definition](#explicit-mapping-definition)
+
+Automatic mapping inference is only supported for [file based virtual schemas](https://github.com/exasol/virtual-schema-common-document-files) using Parquet and CSV files. All other virtual schemas and file formats require an explicit mapping definition.
+
+### Automatic Mapping Inference
+
+To use automatic mapping inference, just omit the `mapping` element from the EDML definition. The virtual schema will then infer the mapping from the schema of the source. Currently this is only supported for Parquet and CSV files.
+
+#### Notes
+
+* The files specified in the `source` must be available when creating the virtual schema. If the files are not available, the `CREATE VIRTUAL SCHEMA` command will fail.
+  * When you don't use automatic mapping inference (i.e. you specify the `mapping` element) you can still create the virtual schema as before without `source` files being available.
+* The adapter will detect the mapping based on the schema of the first file. Please make sure that all files specified as `source` are using the same schema, else the mapping may be wrong.
+* The adapter will detect the mapping when the virtual schema is created. If the schema of the `source` files changes, please drop and re-create the virtual schema to run the auto-inference again.
+* Creating the virtual schema with auto-inference will take longer because the adapter needs to read files from the `source`.
```

Comment:
```suggestion
* Creating the virtual schema with auto-inference will take longer because the adapter needs to read files from the `source` in order to infer the mapping.
```